assets/js/components/common/Mapbox.jsx

```diff
@@ -1,183 +1,28 @@
-import React, { Component } from 'react'
-import { bindActionCreators } from 'redux'
-import { connect } from 'react-redux'
-import { push } from 'connected-react-router'
-import mapboxgl from 'mapbox-gl'
-import { parseLocation } from '../../util/geolocation'
-
-@connect(null, mapDispatchToProps)
-class Mapbox extends Component {
-  componentDidMount() {
-    this.setMapDefaults()
-
-    this.map.on('load', () => {
-      this.addGatewaysToMap()
-      this.addPopupsToMap()
-
-      if (this.props.view === "index") {
-        this.map.scrollZoom.enable()
-      }
-    })
-  }
-
-  componentDidUpdate(prevProps) {
-    if (prevProps.gateways.length !== this.props.gateways.length) {
-      this.removeGatewaysFromMap()
-      this.addGatewaysToMap()
-    }
-  }
-
-  setMapDefaults = () => {
-    mapboxgl.accessToken = 'pk.eyJ1IjoiYWxsZW5hbiIsImEiOiJjajNtNTF0Z2QwMDBkMzdsNngzbW4wczJkIn0.vLlTjNry3kcFE7zgXeHNzQ'
-
-    let initialCenter = [-93.436, 37.778] //US Center
-    let initialZoom = 2
-    if (this.props.view === "show") {
-      initialCenter = [this.props.gateways[0].longitude, this.props.gateways[0].latitude]
-      initialZoom = 13
-    }
-
-    this.map = new mapboxgl.Map({
-      container: this.mapContainer,
+import React, { useRef, useEffect, useState } from 'react'
+import mapboxgl from '!mapbox-gl'
+
+mapboxgl.accessToken = 'pk.eyJ1IjoicGV0ZXJtYWluIiwiYSI6ImNrczZkZTByazA4eWwycHA1OTV4Mml1OWcifQ.WRdWVfa35mv_9VogDL40xw'
```

Comment:
should this be configurable w/ `.env` so that open source users can set their own since they won't be able to use this one?

Reply:
great point